app/src/main/java/com/cmput301f20t21/bookfriends/ui/component/BaseBookListAdapter.java

```diff
@@ -15,22 +24,25 @@
 
 import java.util.List;
 
-public abstract class BaseBookListAdapter<T extends Book> extends RecyclerView.Adapter<BaseBookListAdapter.ViewHolder<T>> {
-    protected List<T> books;
+/**
+ * A abstract adapter used for all book lists
+ */
+public abstract class BaseBookListAdapter extends RecyclerView.Adapter<BaseBookListAdapter.ViewHolder> {
+    protected List<Book> books;
```

Comment:
removed the generic here to get rid of a lot of warnings on ListAdapters

the generic was initially here because we had "AvailableBook" entity before